node/metrics.go

```diff
@@ -225,12 +227,25 @@ func (g *Metrics) collectOnchainMetrics() {
 			})
 			for i, op := range operatorStakeShares {
 				if op.operatorId == g.operatorId {
+					g.allQuorumCache[q] = true
 					g.RegisteredQuorumsStakeShare.WithLabelValues(fmt.Sprintf("%d", q)).Set(op.stakeShare)
 					g.RegisteredQuorumsRank.WithLabelValues(fmt.Sprintf("%d", q)).Set(float64(i + 1))
 					g.logger.Info("Current operator registration onchain", "operatorId", g.operatorId.Hex(), "blockNumber", blockNum, "quorumId", q, "stakeShare (basis point)", op.stakeShare, "rank", i+1)
 					break
 				}
 			}
 		}
+		// Check if operator deregistered for an existing quorum, set the stake share and rank to 0
```

Comment:
ok I separated that function out and called for that case too. I feel for other cases, where we have an error we don't have to reset quorum since we don't know what is the state

Reply:
c1446780834ae38898267cdd15709826c19df56b

Reply:
Another thing is that the loop may hit other errors (e.g. not able to get the state), then it'll skip the update for the metric, will that cause any issuse?

Reply:
that's what I mentioned above. If it's not able to get the stake, then we probably shouldn't update and let it give another try later. I think it's fine choice for now. we can ask for feedback later.